syft/pkg/cataloger/golang/licenses.go

```diff
@@ -1,29 +1,88 @@
 package golang
 
 import (
+	"archive/zip"
+	"bytes"
 	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"net/url"
 	"os"
 	"path"
+	"path/filepath"
 	"regexp"
 	"strings"
 
 	"github.com/mitchellh/go-homedir"
+	"gopkg.in/src-d/go-billy.v4/memfs"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/storage/memory"
 
 	"github.com/anchore/syft/internal/licenses"
 	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/event"
 	"github.com/anchore/syft/syft/source"
 )
 
+const (
+	defaultRemoteProxies = "https://proxy.golang.org,direct"
+	directProxyOnly      = "direct"
+)
+
 type goLicenses struct {
-	searchLocalModCacheLicenses bool
-	localModCacheResolver       source.FileResolver
+	opts                  GoCatalogerOpts
+	localModCacheResolver source.WritableFileResolver
+	progress              *event.GenericProgress
 }
 
 func newGoLicenses(opts GoCatalogerOpts) goLicenses {
 	return goLicenses{
-		searchLocalModCacheLicenses: opts.SearchLocalModCacheLicenses,
-		localModCacheResolver:       modCacheResolver(opts.LocalModCacheDir),
+		opts:                  opts,
+		localModCacheResolver: modCacheResolver(opts.LocalModCacheDir),
+		progress: &event.GenericProgress{
+			SubStatus:          true,
+			RemoveOnCompletion: true,
+			Title:              "Downloading go mod",
+		},
+	}
+}
+
+func remoteProxies(proxy string, noProxy string, module string) (proxies []string) {
+	if proxy == "" {
+		proxy = os.Getenv("GOPROXY")
```

Comment:
env variables should be evaluated as early as possible at runtime -- this also looks like it would be reevaluated several times during cataloging.

Could this get pushed all the way up to application config processing? Or somewhere above this level of processing ? 

(same comment for other env evaluations)